### PR TITLE
fix: [CDS-59209]: displaying infra details even when environment id and name are different

### PR DESCRIPTION
**Pull request by satyam-harness**
#### Screenshots
before
image

after
image

NA

#### PR Checklist

<details>
<summary>Please check if your PR fulfils the following requirements</summary>

- [ ] Tests for the changes have been added. Ideally, include a test that fails without this PR but passes with it.
- [ ] Docs have been [added/updated](https://harness.atlassian.net/jira/software/c/projects/DOC/boards/40).
</details>

<details>
<summary>Use the following comments to re-trigger PR Checks</summary>

- Jest: `retrigger jest`
- Prettier: `retrigger prettier`
- Type Check: `retrigger typecheck`
- ESLint: `retrigger eslint`
- Sonar: `retrigger sonar`
- Standards: `retrigger standards`
- Build: `retrigger build`
- Title Check: `retrigger titlecheck`
- Feature Name Check: `trigger featurenamecheck`
- Coverage: `retrigger coverage`
- Rebase: `trigger rebase`
- Cypress Rest: `retrigger cypress-rest`
- Cypress CD: `retrigger cypress-cd`
- Cypress Pipeline: `retrigger cypress-pipeline`
- Cypress CV: `retrigger cypress-cv`
- Fix Prettier: `fix prettier`
</details>

### CODE REVIEW

**Comment by satyam-harness** [comment]
retrigger jest

**Comment by satyam-harness** [comment]
retrigger coverage

File: src/modules/75-cd/components/PipelineSteps/DeployEnvironmentEntityStep/EnvironmentEntitiesList/EnvironmentEntityCard/EnvironmentEntityCard.tsx [hunk 1]
```diff
@@ -73,7 +73,7 @@ const getScopedRefUsingIdentifier = (
     return envRef
   }
   return get(values, 'environments', []).find(
-    (env: SelectOption) => getIdentifierFromName(env.label) === environment?.identifier
+    (env: SelectOption) => getIdentifierFromName(env.value as string) === environment?.identifier
```
Comment (ishantmehtaharness) [inline src/modules/75-cd/components/PipelineSteps/DeployEnvironmentEntityStep/EnvironmentEntitiesList/EnvironmentEntityCard/EnvironmentEntityCard.tsx:76]: cc: @AshwinBhatkal-Harness @vikrant-harness

**Comment by satyam-harness** [comment]
retrigger jest

**Review by AshwinBhatkal-Harness — APPROVED** [review]

**Review by vikrant-harness — APPROVED** [review]

**Comment by satyam-harness** [comment]
retrigger titlecheck

**Review by YounixIO — APPROVED** [review]

**Pull request merged**